cmd/crypto/key.go

```diff
@@ -45,7 +45,7 @@ func GenerateKey(extKey [32]byte, random io.Reader) (key ObjectKey) {
 		logger.CriticalIf(context.Background(), errOutOfEntropy)
 	}
 	sha := sha256.New()
-	sha.Write(extKey[:])
+	sha.Write(extKey)
```

Comment:
Shouldn't we check len(extKey)?

Reply:
Added a check. However, all of this needs further refactoring. Needs to be done separately